docs/api/explainer.md

```diff
@@ -0,0 +1 @@
+::: explainer
```

Comment:
[Here](https://tompham97.github.io/diffusers-interpret/api/explainer/)'s an example of what mkdocstrings has pulled from `explainer.py`.

Reply:
not a bad start! I like the documentation style. We can definitely adapt the code formatting and docstrings for them to look better in the documentation.
Do you know why other methods were not imported to the documentation? For instance, [the __init__ method](https://github.com/JoaoLages/diffusers-interpret/blob/2dd01d6a494dd0bc18b5c2fa99dda7132ae03f42/src/diffusers_interpret/explainer.py#L27), [gradient checkpointing](https://github.com/JoaoLages/diffusers-interpret/blob/2dd01d6a494dd0bc18b5c2fa99dda7132ae03f42/src/diffusers_interpret/explainer.py#L333) or [_mimic_pipeline_call](https://github.com/JoaoLages/diffusers-interpret/blob/2dd01d6a494dd0bc18b5c2fa99dda7132ae03f42/src/diffusers_interpret/explainer.py#L349). `_mimic_pipeline_call` is actually a private method, so idk if we should add it to the docs, need to see what are the common practices

Reply:
I tried to specify the functions by calling `::: explainer.BasePipelineExplainer.__init__` but it didn't work as intended. I'm looking into the issue but will need another day or two before I could update you with the findings ☺️

Reply:
> `_mimic_pipeline_call` is actually a private method, so idk if we should add it to the docs, need to check what are the common practices

I'll actually need to study public and private methods, as well as related terminologies. I'm still learning Python as a beginner and would appreciate any of your recommendations for resource 😅

Reply:
private methods don't really exist in Python, it's just a convention from other Object-oriented programming languages as Java

Reply:
> Do you know why other methods were not imported to the documentation? For instance, [the **init** method](https://github.com/JoaoLages/diffusers-interpret/blob/2dd01d6a494dd0bc18b5c2fa99dda7132ae03f42/src/diffusers_interpret/explainer.py#L27), [gradient checkpointing](https://github.com/JoaoLages/diffusers-interpret/blob/2dd01d6a494dd0bc18b5c2fa99dda7132ae03f42/src/diffusers_interpret/explainer.py#L333) or [_mimic_pipeline_call](https://github.com/JoaoLages/diffusers-interpret/blob/2dd01d6a494dd0bc18b5c2fa99dda7132ae03f42/src/diffusers_interpret/explainer.py#L349).

So a method needs to have an appropriately-formated docstring in order to be imported into the documentation. Here are a few things that I've discovered thus far:
- The docstring cannot be just empty pairs of quotations `''' '''`.
- Any sentences in the beginning of a docstring will be formatted as [the description](https://tompham97.github.io/diffusers-interpret/api/explainer/#explainer.BasePipelineImg2ImgExplainer:~:text=%C2%B6-,Calls%20a%20DiffusionPipeline%20and%20generates%20explanations%20for%20a%20given%20prompt.,-Parameters%3A).
- Arguments included under `Args: ` will show up in a [table of parameters](https://tompham97.github.io/diffusers-interpret/api/explainer/#explainer.BasePipelineImg2ImgExplainer:~:text=Parameters%3A,%7B%7D).
- Likewise, things under `Returns: ` will show up in the [returns table](https://tompham97.github.io/diffusers-interpret/api/explainer/#explainer.BasePipelineImg2ImgExplainer:~:text=%7B%7D-,Returns%3A,%5D%20if%20init_image%20is%20not%20None%20and%20explanation_2d_bounding_box%20is%20not%20None,-Source%20code%20in)
- Private methods with a single underscore in the front will not be included.
- Methods are listed alphabetically, with characters such as `_` being listed first. For instance, `__call__` would be before `__init__`, which is before `def_...`.
- If there's a specific `@` that belongs to a method, it would also be included in the documentation.

So long story short, you just need to include docstrings in the correct format in order for methods like `__init__` to appear. What do you think about my explanation? Does it make sense to you?

Reply:
Yup, totally makes sense! 👍